include/deal.II/fe/component_mask.h

```diff
@@ -204,7 +204,8 @@ class ComponentMask
   /**
    * Exception
    */
-  DeclException0 (ExcNoComponentSelected);
+  DeclExceptionMsg (ExcNoComponentSelected, "The number of selected
+                    components in a mask must be greater than zero.");
```

Comment:
You use a multiline string `"...\n..."`. This violates the C++ standard, although it is accepted by some compilers. The way you should write this is
```
  DeclExceptionMsg (Exc...,
                                  "abc "
                                  "def");
```
C++ says that multiple strings enclosed in double quotes will simply be collated into one, which is here `"abc def"`. This also explains why I have a space at the end of the first string: I simply want this to be a space between individual words.